File: Daily Exercises/Mar 2021/1047. Remove All Adjacent Duplicates In String/1047. Remove All Adjacent Duplicates In String.py
class Solution:
    def removeDuplicates(self, S: str) -> str:
        stack=list()  # use list as a stack
        for item in S:
            if stack and stack[-1]==item:
                stack.pop()  # cancel it out
            else:
                stack.append(item)
        return ''.join(stack)

File: Daily Exercises/Mar 2021/1047. Remove All Adjacent Duplicates In String/1047. Remove All Adjacent Duplicates In String.cpp
class Solution {
public:
    string removeDuplicates(string S) {
        string s;   // use string as a stack
        for(auto &item:S){
            if(!s.empty() and s.back()==item){ 
                s.pop_back();  // cancel it out
            }else{
                s.push_back(item);
            }
        }
        return s;
    }
};
